src/theory/quantifiers/sygus/sygus_repair_const.cpp

```diff
@@ -206,68 +207,107 @@ bool SygusRepairConst::repairSolution(const std::vector<Node>& candidates,
                                   << std::endl;
       return false;
     }
+  }
 
-    // do miniscoping explicitly
-    if (fo_body[1].getKind() == AND)
+  Trace("cegqi-engine") << "Repairing previous solution..." << std::endl;
+  // make the satisfiability query
+  bool needExport = true;
```

Comment:
Can you encapsulate this block of code so that we don't pollute this utility?  I actually did this on this branch: https://github.com/CVC4/CVC4/pull/2340/files.